### PR TITLE
Use ap-auth-sidecar

### PR DESCRIPTION
**Pull request by pgvishnuram**
## Description

Replace nginx unprivileged image with ap-authsidecar 

## Related Issues

NA

## Testing

NA

## Merging

merge this change to master

### CODE REVIEW

**Review by andriisoldatenko — APPROVED** [review]

**Comment by pgvishnuram** [comment]
@vishwas-astro

File: tests/chart_tests/test_authsidecar.py [hunk 1]
```diff
@@ -156,7 +164,11 @@ def test_authSidecar_houston_configmap_with_annotation(self, kube_version):
             kube_version=kube_version,
             values={
                 "global": {
-                    "authSidecar": {"enabled": True},
+                    "authSidecar": {
+                        "enabled": True,
+                        "repository": "nginxinc/nginx-unprivileged",
```
Comment (danielhoherd) [inline tests/chart_tests/test_authsidecar.py:169]: If we're going to set this then there should be an assertion against it.

Comment (pgvishnuram) [reply tests/chart_tests/test_authsidecar.py:169]: okay will make those changes

File: tests/chart_tests/test_authsidecar.py [hunk 2]
```diff
@@ -126,7 +126,15 @@ def test_authSidecar_houston_configmap_without_annotation(self, kube_version):
         """Test Houston Configmap with authSidecar."""
         docs = render_chart(
             kube_version=kube_version,
-            values={"global": {"authSidecar": {"enabled": True}}},
+            values={
+                "global": {
+                    "authSidecar": {
+                        "enabled": True,
+                        "repository": "quay.io/astronomer/ap-auth-sidecar",
```
Comment (danielhoherd) [inline tests/chart_tests/test_authsidecar.py:133]: Can we make this something that is nothing like what we would actually use so when it fails it's clear that the failure isn't related to the defaults?

Comment (danielhoherd) [inline tests/chart_tests/test_authsidecar.py:133]:
```suggestion
                        "repository": "some/custom/repository",
```

File: tests/chart_tests/test_authsidecar.py [hunk 3]
```diff
@@ -126,7 +126,15 @@ def test_authSidecar_houston_configmap_without_annotation(self, kube_version):
...
+                        "tag": "1.21.6-2",
```
Comment (danielhoherd) [inline tests/chart_tests/test_authsidecar.py:134]:
```suggestion
                        "tag": "custom-tag",
```

**Review by rob-1126 — APPROVED** [review]

**Comment by rob-1126** [comment]
@pgvishnuram I've got a similar PR open at https://github.com/astronomer/astronomer/pull/1337 - I'm OK with this one but that one fixes a bug that will prevent users with custom repositories from over-riding the image name that probably would be useful a number of places in our chart. It certainly won't keep me up at night if you decide to go with this one.

**Review by danielhoherd — APPROVED** [review]

**Pull request merged**